apache/incubator-zeppelin
# ZEPPELIN-825: Set appropriate default permissions and allow user to c…

### What is this PR for?
This PR makes users who set only r/w permissions the effective owners of the note.
1. If you set readers, and writers and owners is empty, it is automatically set to the user requesting the change.
2. If you set writers, and owners is empty, it is set it to the user requesting the change.
It also fixes a bug that did not allow one to clear the permissions for a note. 

### What type of PR is it?
Improvement 

### Todos

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-818

### How should this be tested?
* Create a new note
* Set readers
* Leave writers and owners empty and click Save.
Expected: Writers and Owners is set to the user requesting the change. 


…lear permissions

This change makes users who set permissions the effective owners of the note.
1. If you set readers, and writers and owners is empty, it is automatically set to the user requesting the change.
2. If you set writers, and owners is empty, it is set it to the user requesting the change.
